dev/linters/yamllint.sh

```diff
@@ -6,9 +6,13 @@ workspace=$(bazel info workspace)
 
 find_args=(
     -not \( -path "${workspace}/deploy/helm/kubecf/charts" -prune \)
-    -path "${workspace}/deploy/helm/kubecf/values.*"
-    -or -not -path "${workspace}/deploy/helm/kubecf/*"
-    -name '*.yaml' -or -name '*.yml'
+    -and
+    \( -path "${workspace}/deploy/helm/kubecf/values.*"
+       -or
+       -not \( -path "${workspace}/deploy/helm/kubecf" -prune \)
```

Comment:
If we pruned `"${workspace}/deploy/helm/kubecf"` how will we ever find `"${workspace}/deploy/helm/kubecf/values.*"`? Isn't that redundant? And wouldn't that no longer find `…/deploy/helm/kubecf/assets/operations/*`?

Note that the old expression did _not_ prune anything.

Reply:
The expression `-path "./deploy/helm/kubecf/values.*" -or -not -path "./deploy/helm/kubecf/*"` does effectively prune all those files too; it just works a bit harder by traversing the whole tree. But the right side of the `-or` will reject all of them.

```
$ find . -path "./deploy/helm/kubecf/values.*" -or -not -path "./deploy/helm/kubecf/*" -name '*.yaml' | grep helm
./deploy/helm/kubecf/values.yaml
```

And this makes sense because the files still have unresolved templating code in them, which makes them invalid YAML. You could only lint the chart if you push it through `helm template` first.

But you are right that the pruning will remove the `values.yaml` from the list, so I have to revert that part.